### PR TITLE
Integrate ENS Record Setting in UI

### LINKED ISSUE #253: Integrate ENS Record Setting Directly into Tabula UI
**Is your feature request related to a problem? Please describe.**
Currently, there needs to be a direct way in the UI for users to set their ENS record. Users have to manually add their publication hash to their ENS record, as shown in the example: [https://app.ens.domains/name/manboy.eth/details](https://app.ens.domains/name/manboy.eth/details). At the same time, they can access it at tabula. gg/#/yourCustomEns, and the app will automatically detect the ENS; the process could be more streamlined.

**Describe the solution you'd like.**
I propose integrating the `multicall` and `setText` methods from the ENS contract [here](https://goerli.etherscan.io/address/0xd7a4f6473f32ac2af804b3686ae8f1932bc35750#writeProxyContract) directly into our UI. Users can easily set their ENS record by adding a button within the publications section. This button will execute all the necessary logic and explain to the user about the underlying processes.

**Additional context**
The goal is to enhance the user experience by simplifying setting ENS records and clarifying what happens in the background.

### PR DESCRIPTION
**Pull request by juliopavila**
## **Description**:

closes #253 

### Problem:
Currently, there's no direct way in the UI for users to set their ENS record. Users have to manually add their publication hash to their ENS record, as shown in the example: [ENS Domain Example](https://app.ens.domains/spiterman.eth?tab=records). While they can access it at `tabula.gg/#/yourCustomEns`, and the app will automatically detect the ENS, the process could be more streamlined.

### Solution:
This PR integrates the `multicall` and `setText` methods from the ENS contract directly into our UI. Users can now easily set their ENS record by adding a button within the publications section. This button will execute all the necessary logic and explain to the user about the underlying processes.

### Changes:
1. **App.tsx**: Imported `EnsProvider` from `./services/ens/context`.
2. **WalletBadge.tsx**: Made several modifications including importing `useEnsContext` and refactoring the logic.
3. **EnsModal.tsx**: New component to handle the ENS modal UI.
4. **SettingSection.tsx**: Added logic to link the publication to the user's ENS name.
5. **ens.context.tsx**: New context for ENS operations.
6. **useEns.ts**: New hook to handle ENS-related operations.

### Additional Context:
The goal of this PR is to enhance the user experience by simplifying the setting of ENS records and clarifying what happens in the background.

---

### **Reviewer Checklist**:
- [ ] Code is clean and follows the project's coding standards.
- [ ] Functionality has been verified and is working as intended.
- [ ] UI/UX is intuitive and matches the design specs.
- [ ] All new dependencies have been declared.
- [ ] PR is linked to the related issue.

### **Screenshots**

image
image
image
image
image

### CODE REVIEW

**Comment by auryn-macmillan** [comment]
Checked out the deploy preview, but it's not showing the new elements to set ENS names.
image

**Review by manboy-eth — COMMENTED** [review]
Nice work, just one note:
We will probably need some way of getting the user to change chain to mainnet, if they are not already on mainnet, since the ENS record is always set on mainnet.

File: packages/app/src/services/ens/hooks/useENS.ts [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+        openNotification({
+          message: "ENS is not supported on this network or an error occurred.",
+          variant: "error",
+        })
+      } finally {
+        setLoading(false)
+      }
+    },
+    [getPublicResolverAddress, openNotification],
+  )
```
Comment (manboy-eth) [inline packages/app/src/services/ens/hooks/useENS.ts:153]: Why do we do a 'multicall' when we are only doing one call?

**Review by cedricwaxwing — APPROVED** [review]
LGTM, nice work Julio!

**Comment by juliopavila** [comment]
Thanks, for your feedback @manboy-eth, I already applied the following changes:

- Changed default to mainnet subgraph, the only "real" ENS registry, as per your suggestion.
- Replaced 'multicall' with 'setText' for a single call operation, cleaned up the code, and tested functionality.

Please review

**Review by manboy-eth — COMMENTED** [review]
Nice, much better :)

But did you do anything to handle this?
> We will probably need some way of getting the user to change chain to mainnet, if they are not already on mainnet, since the ENS record is always set on mainnet.

When the user for instance has a publication on Gnosis Chain, the user must still set the ENS record on mainnet.

File: packages/app/src/components/commons/NetworkModal.tsx [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+  top: "50%",
+  left: "50%",
+  transform: "translate(-50%, -50%)",
+  borderRadius: 8,
+  width: 648,
+  background: palette.whites[1000],
+  padding: 24,
+})
+
+const NetworkModal: React.FC<any> = ({ open, handleClose }) => {
```
Comment (manboy-eth) [inline packages/app/src/components/commons/NetworkModal.tsx:26]: Missing props type.

File: packages/app/src/components/commons/NetworkModal.tsx [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+  const openNotification = useNotification()
+  const { library } = useWeb3React()
+  const [loading, setLoading] = useState<boolean>(false)
+
+  const handleSwitch = async () => {
+    if (library) {
+      try {
+        setLoading(true)
+        await library.send("wallet_switchEthereumChain", [{ chainId: "0x1" }])
+        handleClose({}, "escapeKeyDown")
```
Comment (manboy-eth) [inline packages/app/src/components/commons/NetworkModal.tsx:37]: Used as MUIs `onClose` ModalProp, but looks like it is actually a `() => void` (from component usage in `packages/app/src/components/views/publication/components/SettingSection.tsx`).

**Review by manboy-eth — CHANGES_REQUESTED** [review]
Excellent, now we just need some minor changes and I think this is ready to merge.

**Pull request merged**